src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2213,7 +2213,20 @@ private BoundExpression BindOutVariableArgument(DeclarationExpressionSyntax decl
 
                 if (isVar)
                 {
-                    return new OutVariablePendingInference(declarationExpression, localSymbol, null);
+                    if (localSymbol.DeclarationKind == LocalDeclarationKind.OutVariable)
```

Comment:
BindArgumentValue should never call this method if the argument is not an Out Var Declaration.

---
In reply to: [103062578](https://github.com/dotnet/roslyn/pull/17380#discussion_r103062578) [](ancestors = 103062578)